lib/elixir/lib/kernel/lexical_tracker.ex

```diff
@@ -171,8 +172,21 @@ defmodule Kernel.LexicalTracker do
   defp add_compile(d, module, :runtime), do: :ets.insert_new(d, {{:mode, module}, :runtime})
   defp add_compile(d, module, :compile), do: :ets.insert(d, {{:mode, module}, :compile})
 
-  defp add_directive(d, module, line, warn, tag) do
+  defp add_directive(d, m_or_mfa, line, warn, :import = tag) when is_atom(m_or_mfa) do
+    # remove possible previous `{m, f, a}` entries, because here the entire module
+    # has been imported instead of selected functions.
+    module = m_or_mfa
+    :ets.select(d, [{{{tag, {module, :"$2", :"$3"}}, :"$4"}, [is_integer: :"$4"], [{{module, :"$2", :"$3"}}]}])
```

Comment:
I think you can use match_delete here: http://erlang.org/doc/man/ets.html#match_delete-2

Something like `:ets.match_delete(d, {tag, {module, _, _}})` maybe? We can just delete all, it doesn't matter if line is integer.